mne/decoding/time_gen.py

```diff
@@ -407,87 +465,114 @@ def _predict_time_loop(X, estimators, cv, slices, predict_mode,
-        _warn_once['vectorization'] = False
-    # Iterate over testing times. If is_single_time_sample, then 1 iteration.
-    y_pred = list()
-    for t, indices in enumerate(slices):
-        # Vectoring chan_times features in case of multiple time samples given
-        # to the estimators.
-        if not is_single_time_sample:
-            X_pred = X[:, :, indices].reshape(n_epochs, -1)
-        else:
+    n_train = len(estimators)
+    n_test = [len(test_times) for test_times in train_times]
+
+    # Loop across training times (i.e. estimators)
+    y_pred = None
+    for train_idx, (estimator_cv, test_times) in enumerate(
+            zip(estimators, train_times)):
+        # Checks whether predict is based on contiguous windows of lengths = 1
+        # time-sample, ranging across the entire times. In this case, we will
+        # be able to vectorize the testing times samples.
+        expected_start = np.arange(n_times)
+        contiguous_start = np.array_equal([sl[0] for sl in test_times],
+                                          expected_start)
+        window_lengths = np.unique([len(sl) for sl in test_times])
+        vectorize_times = (window_lengths == 1) and contiguous_start
+        if vectorize_times:
+            # In vectorize mode, we avoid iterating over time test_times.
+            test_times = [slice(expected_start[0], expected_start[-1] + 1, 1)]
+        elif _warn_once.get('vectorization', True):
+            warn('not vectorizing predictions across testing times, using a '
+                 'time window with length > 1')
+            _warn_once['vectorization'] = False
+
+        # Iterate over testing times. If vectorize_times: 1 iteration.
+        for test_idx, test_time in enumerate(test_times):
+            # Vectoring chan_times features in case of multiple time samples
+            # given to the estimators.
             X_pred = X
-
-        if predict_mode == 'mean-prediction':
-            # Predict with each fold's estimator and average predictions.
-            y_pred.append(_predict(X_pred, estimators,
-                          is_single_time_sample=is_single_time_sample,
-                          predict_method=predict_method))
-        elif predict_mode == 'cross-validation':
-            # Predict with the estimator trained on the separate training set.
-            for k, (train, test) in enumerate(cv):
-                if test.size == 0:
-                    continue
-                # Single trial predictions
-                X_pred_t = X_pred[test_epochs_slices[k]]
-                # If is_single_time_sample, we are predicting each time sample
-                # as if it was a different epoch (vectoring)
-                y_pred_ = _predict(X_pred_t, estimators[k:k + 1],
-                                   is_single_time_sample=is_single_time_sample,
+            if not vectorize_times:
+                X_pred = X[:, :, test_time].reshape(n_epochs, -1)
```

Comment:
if we are indexing we typically name the variable so that it ends with `idx`. Right now, you have `test_time` and `test_idx` and I can't make out the difference between them. At first, I thought one was float and couldn't be used for indexing, but now I see it's being used for indexing